speech/recognize_streaming.js

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var async = require('async');
+var fs = require('fs');
+var path = require('path');
+var grpc = require('grpc');
+var googleProtoFiles = require('google-proto-files');
+var googleAuth = require('google-auto-auth');
+
+// [START proto]
+var PROTO_ROOT_DIR = googleProtoFiles('..');
+var PROTO_FILE_PATH = googleProtoFiles('cloud', 'speech', 'v1', 'cloud_speech.proto');
+
+var protoDescriptor = grpc.load({
+  root: PROTO_ROOT_DIR,
+  file: path.relative(PROTO_ROOT_DIR, PROTO_FILE_PATH)
+}, 'proto', {
+  binaryAsBase64: true,
+  convertFieldsToCamelCase: true
+});
+var speechProto = protoDescriptor.google.cloud.speech.v1;
+// [END proto]
+
+// [START authenticating]
+function getSpeechService (callback) {
+  var authClient = googleAuth({
+    scopes: [
+      'https://www.googleapis.com/auth/cloud-platform'
+    ]
+  });
+
+  authClient.getAuthClient(function (err, authClient) {
```

Comment:
doesn't `authClient` alias a variable in the outer scope? Should we give it a different name?